docs/guide/core-concepts.rst

```diff
@@ -13,364 +13,489 @@ alike. If GIS is brand, brand new to you, `this
 document <https://www.gislounge.com/what-is-gis/>`__ is a useful high
 level overview.
 
-Basic Terms
-===========
+Glossary
+========
 
--  **Tile:** A grid of numeric *cells* that represent some data on the
-   Earth.
--  **Cell:** A single unit of data in some grid, also called a
-   *Location* in GIS.
--  **Layer:** or "Tile Layer", this is a grid (or cube) of *Tiles*.
--  **Zoom Layer:** a *Tile Layer* at some zoom level.
--  **Key:** Used to index a *Tile* in a grid (or cube) of them.
--  **Key Index:** Used to transform higher-dimensional *Keys* into one
-   dimension.
--  **Metadata:** or "Layer Metadata", stores information critical to
-   Tile Layer IO.
--  **Layout Definition:** A description of a Tile grid (its dimensions,
-   etc).
--  **Extent:** or "Bounding Box", represents some area on the Earth.
--  **Raster:** A *Tile* with an *Extent*.
--  **Vector:** or "Geometry", these are Point, Line, and Polygon data.
--  **Feature:** A *Geometry* with some associated metadata.
--  **RDD:** "Resilient Distributed Datasets" from `Apache
-   Spark <http://spark.apache.org/>`__. Can be thought of as a highly
-   distributed Scala ``Seq``.
+- `Vector <#geometries>`_ or **Geometry**: Structures built up by
+  connecting Points in space; includes ``Point``\s, ``Line``\s,
+  ``Polygon``\s.
+- `Extent <#extents>`_ or **Bounding Box**: An axis aligned, rectangular
+  region.
+- `Feature <#features>`_: A *Geometry* with some associated metadata.
+- `Cell <#working-with-cell-values>`_: A single unit of data in some
+  grid.
+- `Tile <#tiles-and-rasters>`_: A grid of numeric *cells* that represent
+  some data on the Earth.
+- `Raster <#rasters>`_: A *Tile* with an *Extent*.
+- **RDD:** "Resilient Distributed Datasets" from `Apache Spark <http://spark.apache.org/>`__. Can be thought of as a distributed Scala ``Seq``.
+- `Key <#keys>`_: Used to index a *Tile* in a grid (or cube) of them.
+- `Key Index <#key-indexes>`_: A means to transform higher-dimensional
```

Comment:
A minor thing, but it might be helpful to have these topics listed in the order in which they appear in the document.